pkg/apis/common/v1/common.go

```diff
@@ -353,6 +353,16 @@ type HasObservedGeneration interface {
 	GetObservedGeneration() int64
 }
 
+// TypeLabelName is used to represent a resource type in k8s resources
+const TypeLabelName = "common.k8s.elastic.co/type"
+
+// HasElasticLabels allows a return of Elastic assigned labels for any object.
+// +kubebuilder:object:generate=false
+type HasElasticLabels interface {
```

Comment:
You mentioned it in the PR description already and naming is hard. I think `Elastic` as a name is maybe not ideal because everything we do is something `Elastic` I guess. I was trying to think of something more descriptive but all seemed pretty tortured like `TypeNameLabels` or incomplete like `TypeLabels`. One option I don't totally hate is `GetIdentityLabels` and `HasIdentityLabels` the rationale being that the name and type labels together identify the resource uniquely e.g. for label selectors and such. Wdyt?

Reply:
I can be fine with that change.  Implemented.